### ISSUE TITLE
Complete or Partial Words

### ISSUE THREAD

**Issue by PDPost** [yiminzme/Multi-Highlight#10]
Hello - thanks for creating this!  I am doing well with it but have a question.  Is it possible to create a checkbox that lets the user highlight only the complete words that are in the search parameters OR highlight any consecutive letters that contain the letters?  For example, I put a "PA" in your Multi-Highlight box looking to highlight only the instances of PA (the state abbreviation for Pennsylvania) however I am seeing "pa" highlighted in words that I don't need to see them highlighted in (paste, park, path, repair, etc).  Having a checkbox that allows the user to switch between "Only" and "Contains" would be very helpful.  Is that something that can be added?  Thanks again!

**Comment by yiminzme** [comment]
Hi PDPost, for now, you can achieve similar effect by (1) changing delimiter to comma "," (2) put PA with a space and comma in the box, i.e. "PA ,".
I will consider adding this feature. Thanks for your suggestion.

**Comment by Mrmohanak** [comment]
problem of highlighting partial words even if i put a space delimiter,
the solution you mentioned of using space the comma is exhausting when using many words.

**Comment by TroyDanielFZ** [comment]
Feature added in last merged commit.

**Comment by yiminzme** [comment]
Thanks, @TroyDanielFZ

**Comment by PDPost** [comment]
Hello,

Thanks so much for adding the changes!

All was working perfectly yesterday but I’m having a problem this morning.

The Highlight box completely erased my entered words, and when I go back to re-enter them it keeps erasing them when I go back.

Are others having this same problem?

Thanks again,
Pat

**Comment by TroyDanielFZ** [comment]
The removal happens when you `Ctrl+Click` the keyword list items, and the keyword list in the edit box content is completely reconstructed from the list. 
I don't know what "when I go back." exactly means. Do you mean, re-open the floating box by clicking the icon?
Will the re-entered keywords show up in the list below?

**Comment by PDPost** [comment]
Hi Troy,

Thanks for your help with this!

Here is what I am seeing this morning…


Yesterday, the Multi Highlight box was populated but, today, all the words are gone and when I go back to re-paste them back in like this…


…I click anywhere outside of the box (like I always did) to “enter’ them but when I come back I instead see this again…


So, I’m not trying to remove (Ctrl+Click) anything but instead trying to re-enter the key words in the search box but they’re not staying.

Should I enter the words a different way?

Thank you,
Pat

**Comment by TroyDanielFZ** [comment]
I commonly do like this.
Do you  check the "Highligh as you type" option?
Which version do you use?

**Comment by PDPost** [comment]
Good morning,

Yes, I did try the “Highlight as you type” option but nothing highlights at all even just temporarily when the word list is pasted in to the box.  When I click out of the box,  

I just now went to “About Chrome” and it says I’m Version 96.0.4664.110 (Official Build) (64-bit).  But,it updated itself and wanted me to “Relaunch”.  After Relaunch, I am not at Version 97.0.4692.71 (Official Build) (64-bit) and I went right to Multi-Highlight, pasted in the words, clicked out of the app to keep them, went back to the app and the words were again gone.

I even clicked “Remove from Chrome” on Multi-Highlight and then added it back in, and then pasted the search words back in but they still disappear/erase after I click out of the box.  I’ve tried this on two computers, too, and it doesn’t work on either of them.

BTW, Multi-Highlight had been working perfectly with this word list since early August through Tuesday and then I I’ve had this problem since yesterday.  I think you did an update overnight and ever since then I haven’t been able to get it to highlight anything.

If it helps, here is my word list…
1GOLD
2PINK census consider data h&b highest improved info intended legal listing mls month online six source three tract twelve value year
3YELLOW boundaries bracket bracketed complexity census conclusion condo consider emphasis engaged hypothetical inspected made man mineral natural ny oh pa planned previous profiler pud reconcil tax weigh wv zon
4BLUE airplane airport beach beltway busy canal cell church comm construction creek driveway dumpster electric expressway fire flight freeway golf grade high highway htew htw hwy indust interstate ladder lagoon lake library line mall ocean overhead parkway path pond power rail repair river road roof rr school shopping street tarp tension tower track train turnpike vacant view waste water wire worship
5GREEN 10% 15% 25% adjustment age amenit as-is basement bath bathroom bed condition condo f.p. f/p finish fireplace fp fpl garage gla gross improved" "as is" land line lot mile net number pool repaired" search site unit
6PURPLE
7CORAL
8RED (air) (describe) attempt bmo borrower breidenstein citizens cunt declin decrease describe homeowner increas interview listed modified not owner price range u.s. unknown xx
10LIGHTRED
11BROWN
12LIGHTPINK

(I figured out back in August what your colors were, so for example “1GOLD” would never highlight but it tells me what color it will be if I enter a word next to it and before the "line break”)

Please let me know if you have any other ideas because I’m really stuck!

Thank you,
Pat
847-275-5671

**Comment by yiminzme** [comment]
@PDPost 
Could you share your setting/configuration (e.g. screenshot)? So we might reproduce the problem you met.
It might be because the delimiter setting is incorrect (by default it is comma ",", in your case you want to use space " ").

Best,

**Comment by TroyDanielFZ** [comment]
I reproduce this problem on my machine, finding several bugs. And I'm correcting thems.

**Comment by PDPost** [comment]
Thanks so much to both of you!

Here are a few screen grabs of all of the settings pages that I could find…




I’ve always left the delimiter at its default because I’m not exactly sure what it does!  But, yes, the space has been fine.  If it works by putting a comma in, though, I’ll certainly do that.

Thanks again very much!
Pat
847-275-5671

**Comment by TroyDanielFZ** [comment]
A [PR](https://github.com/yiminzme/Multi-Highlight/pull/19#issue-1095415173) created. 
Within limited tests, the problem mentioned above is solved.

**Comment by PDPost** [comment]
Hi guys,

That’s great - thank you!

I just copied and pasted my word list but they are still not staying.  Should it be working yet or will you be sending out an update to the app?

Thanks again very much,
Pat

**Comment by TroyDanielFZ** [comment]
> Hi guys, That’s great - thank you! I just copied and pasted my word list but they are still not staying. Should it be working yet or will you be sending out an update to the app? Thanks again very much, Pat

Thanks for testing.
Do you clone the newly uploaded code?
You can clone it, and turn on the `developer mode` in the extension page extensions: chrome://extensions/, and load unpacked to open the 'src' folder.  
Thanks very much.

**Comment by PDPost** [comment]
Hi guys,

Well, I’ve figured out how to turn on developer mode and I see where it says “Load Unpacked” but nothing is opening.

I’ve also tried Removing Multi Highlight and adding it back in to see if that would include the fixes but that doesn’t work either.

I do see that the Version is 2.7.

Please let me know how to do this - thank you!
Pat

**Comment by TroyDanielFZ** [comment]
@PDPost 
After clicking “Load Unpacked”, should navigate to the cloned folder, and choose the `src` folder.
I just merged the PR, so, you can re-clone it from `https://github.com/yiminzme/Multi-Highlight.git`. 
Right now, the version should be 2.7.1.

**Comment by PDPost** [comment]
Hi guys,

Thank you - I forgot to reply because, about an hour after I eMailed you that it still wasn’t working, I tried it again and it worked!  So, maybe it just needed some time.

I was able to re-enter all of my word list and see how they now appear at the bottom of the window.  If I get some time over the weekend, I am going to play around some more with the Delimiter (which I changed from nothing to a comma) and then mix words like “as” and “is” so that it doesn’t highlight both in the entire document any more but instead only when “as is” are together.  I am also going to play around with “Only highlight whole words” because I should be able to minimize a lot of random words that didn’t need to be highlighted down to just the words that I am specifically looking for with your new Version 2.7 features.

So, thanks again to both of you very much!
Pat
